internal/memoize/nosync.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build tinygo && memoize_builders
+
+package memoize
+
+import "sync"
+
+var doer = makeDoer(new(sync.Map))
+
+// Do executes and returns the results of the given function, unless there was a cached
+// value of the same key. Only one execution is in-flight for a given key at a time.
+// The boolean return value indicates whether v was previously stored.
+func Do(key string, fn func() (interface{}, error)) (interface{}, error) {
+	value, err, _ := doer(key, fn)
+	return value, err
+}
+
+// makeDoer returns a function that executes and returns the results of the given function
+func makeDoer(cache *sync.Map) func(string, func() (interface{}, error)) (interface{}, error, bool) {
```

Comment:
Sorry I know I suggested copying the same PR. But realized that since this is wasilibs we shouldn't use any of these sync mechanisms - there are no threads (maybe in the future but I'd rather deal with it when it happens). So it'd be good to simplify the implementation and tests to just use a normal map and no other libraries.

Reply:
I had the same thought about the sync package but then realise people could
still want to use wasilibs with go, and hence the sync isn't? Still we can
remove it for now until someone requests for it.

Reply:
Yeah that's probably fine. I have bad memories of a sync pool not actually pooling for some time in TinyGo and have some caution towards that package but if sync.Map is mostly a normal map in practice it should be ok